spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/zookeeper/ZooKeeperStreamDefinitionRepository.java

```diff
@@ -154,8 +155,10 @@ public StreamDefinition findOne(String id) {
 			}
 			Map<String, String> map = ZooKeeperUtils.bytesToMap(bytes);
 			StreamDefinition streamDefinition = new StreamDefinition(id, map.get(DEFINITION_KEY));
-			List<ModuleDefinition> moduleDefinitions = objectReader.readValue(map.get(MODULE_DEFINITIONS_KEY));
-			streamDefinition.setModuleDefinitions(moduleDefinitions);
+			if (map.get(MODULE_DEFINITIONS_KEY) != null) {
```

Comment:
The value will be null here only if the stream definition was stored using the previous version of XD. In other cases, we always set the module definitions based on the fix for XD-2731. This method handles the find() method only. In case of other loadStream operations, the null check here https://github.com/spring-projects/spring-xd/pull/1515#discussion_r26995231 would address it.